### PR TITLE
Update Sugar Controller docs

### LINKED ISSUE #4896: Eventing Sugar Controller Integration
## What is changing? (Please include as many details as possible.)
The sugar reconciler has been integrated into the base eventing controller and is now controlled by two LabelSelector fields.

## How will this impact our users?
The Sugar controller can be used much more flexibly with the changes. The Sugar controller's legacy behaviour can be replicated, however the steps to do that will change.
The changes also require an update to our [Sugar Controller Installation & Usage docs](https://knative.dev/docs/eventing/sugar/#trigger-example)

## In what release will this take happen (to the best of your knowledge)?
v1.4

## Context
Link to associated PRs or issues from other repos here.
1. PR - https://github.com/knative/eventing/pull/6027

### PR DESCRIPTION
**Pull request by xtreme-sameer-vohra**
Fixes #4896 

## Proposed Changes 
- Update existing Sugar Controller page to reference LabelSelector Changes
- Add Configuration page for "config-sugar" ConfigMap

### CODE REVIEW

**Comment by xtreme-sameer-vohra** [comment]
Hmm not sure why the `yaml` block is not rendering properly on https://deploy-preview-4925--knative.netlify.app/docs/eventing/sugar/#automatic-broker-creation

which has
```
	  ```yaml
	  apiVersion: v1
	  kind: ConfigMap
	  metadata:
	  name: config-sugar
	  namespace: knative-eventing
	  labels:
	      eventing.knative.dev/release: devel
	  data:
	    # Use an empty object to enable for all namespaces
	    namespace-selector: |
	      {}
	  
	    # Use an empty object to enable for all triggers
	    trigger-selector: |
	      {}
	  ```
```

File: docs/eventing/configuration/sugar-configuration.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          values: ["enabled"]
+
+      # Use an empty object to enable for selected triggers
+      trigger-selector: |
+        matchExpressions:
+        - key: "eventing.knative.dev/injection"
+          operator: "In"
+          values: ["enabled"]
+    ```
+
```
Comment (csantanapr) [inline docs/eventing/configuration/sugar-configuration.md:39]: In this case, don't indent 
```suggestion

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
name: config-sugar
namespace: knative-eventing
labels:
    eventing.knative.dev/release: devel
data:
  # Use an empty object to enable for selected namespaces
  namespace-selector: |
    matchExpressions:
    - key: "eventing.knative.dev/injection"
      operator: "In"
      values: ["enabled"]

  # Use an empty object to enable for selected triggers
  trigger-selector: |
    matchExpressions:
    - key: "eventing.knative.dev/injection"
      operator: "In"
      values: ["enabled"]
```

```

File: docs/eventing/sugar/README.md [hunk 2]
```diff
@@ -38,16 +35,33 @@ the default settings:
...
+        eventing.knative.dev/release: devel
+    data:
+      # Use an empty object to enable for all namespaces
+      namespace-selector: |
+        {}
+
+      # Use an empty object to enable for all triggers
+      trigger-selector: |
+        {}
+    ```
```
Comment (csantanapr) [inline docs/eventing/sugar/README.md:58]: In this case don't ident
```suggestion
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
name: config-sugar
namespace: knative-eventing
labels:
    eventing.knative.dev/release: devel
data:
  # Use an empty object to enable for all namespaces
  namespace-selector: |
    {}

  # Use an empty object to enable for all triggers
  trigger-selector: |
    {}
```

```

File: docs/eventing/configuration/sugar-configuration.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Configure Sugar Controller
```
Comment (csantanapr) [inline docs/eventing/configuration/sugar-configuration.md:1]: You need to add this to the navigation menu [nav.yaml](https://github.com/knative/docs/blob/main/config/nav.yml#L208)

File: docs/eventing/configuration/sugar-configuration.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+This topic describes how to configure the Sugar Controller. You can configure the Sugar controller to create a Broker when a Namespace or Trigger is created with configured labels. See [Knative Eventing Sugar Controller](../sugar/README.md) for an example.
+
+
+The default `config-sugar` ConfigMap disables Sugar Controller.
```
Comment (csantanapr) [inline docs/eventing/configuration/sugar-configuration.md:6]:
```suggestion
The default `config-sugar` ConfigMap disables Sugar Controller, by setting `namespace-selector` and `trigger-selector` to an empty string.
```

**Review by csantanapr — CHANGES_REQUESTED** [review]

File: docs/eventing/configuration/sugar-configuration.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    ```yaml
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+    name: config-sugar
+    namespace: knative-eventing
+    labels:
+        eventing.knative.dev/release: devel
+    data:
+      # Use an empty object to enable for selected namespaces
```
Comment (csantanapr) [inline docs/eventing/configuration/sugar-configuration.md:25]: This comment is confusing. An empty object will enable it for all namespaces not selected namespaces.

File: docs/eventing/configuration/sugar-configuration.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+      namespace-selector: |
+        matchExpressions:
+        - key: "eventing.knative.dev/injection"
+          operator: "In"
+          values: ["enabled"]
+
+      # Use an empty object to enable for selected triggers
```
Comment (csantanapr) [inline docs/eventing/configuration/sugar-configuration.md:32]: Same here, empty object `{}` will enable it for all triggers, in this case, you are being selective about which triggers this feature applies.

File: docs/eventing/sugar/README.md [hunk 7]
```diff
@@ -38,16 +35,33 @@ the default settings:
 
 There might be cases where automated Broker creation is desirable, such as on
 namespace creation, or on Trigger creation. The Sugar controller enables those
-use-cases:
+use-cases. The following sample configuration of the `sugar-config` ConfigMap
```
Comment (csantanapr) [inline docs/eventing/sugar/README.md:38]: Do we really think this should be the default to be documented? If it is then mention something about the option to use a label to restrict where the behavior is applied by using the global configmap `config-sugar`

File: docs/eventing/configuration/sugar-configuration.md (hunk 5 above)
Comment (xtreme-sameer-vohra) [reply docs/eventing/configuration/sugar-configuration.md]: good catch, thats an oversight

File: docs/eventing/configuration/sugar-configuration.md (hunk 6 above)
Comment (xtreme-sameer-vohra) [reply docs/eventing/configuration/sugar-configuration.md]: good catch, thats an oversight

File: docs/eventing/sugar/README.md (hunk 7 above)
Comment (xtreme-sameer-vohra) [reply docs/eventing/sugar/README.md:38]: Thanks for the feedback, added a label selector for the namespaces which would limit the application to be more intentional. The Trigger example shows how to unleash it everywhere.

Comment (xtreme-sameer-vohra) [reply docs/eventing/sugar/README.md:38]: Also fixed some other inconsistencies that I had missed earlier.

**Review by csantanapr — APPROVED** [review]
Thanks for the updates @xtreme-sameer-vohra

**Pull request merged**